python/cudf/cudf/core/frame.py

```diff
@@ -1202,6 +1204,171 @@ def __arrow_array__(self, type=None):
+            Default = 1 if frac = None.
+        frac : float, optional
+            Fraction of axis items to return. Cannot be used with n.
+        replace : bool, default False
+            Allow or disallow sampling of the same row more than once.
+            replace == True is not yet supported for axis = 1/"columns"
+        weights : str or ndarray-like, optional
+            Only supported for axis=1/"columns"
+        random_state : int or None, default None
+            Seed for the random number generator (if int), or None.
+            If None, a random seed will be chosen.
+        axis : {0 or ‘index’, 1 or ‘columns’, None}, default None
+            Axis to sample. Accepts axis number or name.
+            Default is stat axis for given data type
+            (0 for Series and DataFrames).
+
+        Returns
+        -------
+        Series or DataFrame
+            A new object of same type as caller containing n items
+            randomly sampled from the caller object.
+
+         Example
+        -------
+        >>> import cudf as cudf
+        >>> df = cudf.DataFrame({"a":{1, 2, 3, 4, 5}})
+        >>> df.sample(3)
+           a
+        1  2
+        3  4
+        0  1
+
+        >>> sr = cudf.Series([1, 2, 3, 4, 5])
+        >>> sr.sample(10, replace=True)
+        1    4
+        3    1
+        2    4
+        0    5
+        0    1
+        4    5
+        4    1
+        0    2
+        0    3
+        3    2
+        dtype: int64
+
+        >>> df = cudf.DataFrame(
+        ... {"a":[1, 2], "b":[2, 3], "c":[3, 4], "d":[4, 5]})
+        >>> df.sample(2, axis=1)
+           a  c
+        0  1  3
+        1  2  4
+        """
+
+        if frac is not None and frac > 1 and not replace:
+            raise ValueError(
+                "Replace has to be set to `True` "
+                "when upsampling the population `frac` > 1."
+            )
+        elif frac is not None and n is not None:
+            raise ValueError(
+                "Please enter a value for `frac` OR `n`, not both"
+            )
+
+        if frac is None and n is None:
+            n = 1
+        elif frac is not None:
+            if axis is None or axis == 0 or axis == "index":
+                n = int(round(self.shape[0] * frac))
+            else:
+                n = int(round(self.shape[1] * frac))
+
+        if axis is None or axis == 0 or axis == "index":
+            if not replace and n > self.shape[0]:
+                raise ValueError(
+                    "Cannot take a larger sample than population "
+                    "when 'replace=False'"
+                )
+
+            if weights is not None:
+                raise NotImplementedError(
+                    "weights is not yet supported for axis=0/index"
+                )
+
+            seed = (
+                random.randint(0, 10000)
+                if random_state is None
+                else np.int64(random_state)
+            )
```

Comment:
Can just use `np.random.randint`, why are we binding from 0 --> 10000?
```suggestion
            seed = np.random.randint(0, 10000, dtype=np.int64)
```

Reply:
There is no default option in np.random.randint to get a random value, so had to provide a range. And thought may be bigger range might slowdown the randint, not sure though.